lib/serviceBuilder.js

```diff
@@ -48,16 +48,18 @@ function statusCodeError(statusCode, allowedStatusCodes, message) {
 function setBodyRelatedHeaders(body, options) {
   const isJSON = !Buffer.isBuffer(body) && typeof body !== 'string' && !(body && typeof body.pipe === 'function')
   if (isJSON) {
-    options.headers = options.headers || {}
-    options.headers['content-type'] = 'application/json;charset=utf-8'
+    options.headers = { 'content-type': 'application/json;charset=utf-8', ...options.headers }
```

Comment:
Like the other comment on this file, you changed the behavior of the function here too